library/std/src/thread/mod.rs

```diff
@@ -698,17 +698,22 @@ where
 }
 
 thread_local! {
+    // Invariant: `CURRENT` and `CURRENT_ID` will always be initialized
+    // together. However, while `CURRENT_ID` will be available during
+    // TLS constructors, `CURRENT` will not.
```

Comment:
I think you mean destructors? And this isn't quite true, key-based TLS is always destroyed (but can be reinitialized, so this isn't that much of a problem).